pgjdbc/src/main/java/org/postgresql/util/PropertyLoader.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import org.postgresql.PGProperty;
+import org.postgresql.util.PSQLException;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+
+public class PropertyLoader {
+  private static final Logger LOGGER = Logger.getLogger("org.postgresql.util.PropertyLogger");
+
+  public static Properties load() throws PSQLException {
+    return load(null);
+  }
+
+  public static Properties load(Properties info) throws PSQLException {
+    return load(System.getenv(), info);
+  }
+
+  public static Properties load(Map<String, String> env, Properties info) throws PSQLException {
```

Comment:
@davecramer @grzm this is the main entry point to build Properties. Actually. having env parameterized is just to inject dependencies to ease unit testing.

This is where the loading order is defined.